### PR TITLE
Make item_assets a dict

### PR DESCRIPTION
**Pull request by kylebarron**
Closes https://github.com/arturo-ai/stac-pydantic/issues/46

### CODE REVIEW

**Comment by vincentsarago** [comment]
@kylebarron can you share an item and add a test ?

**Comment by kylebarron** [comment]
Test added

File: tests/test_models.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ class Config:
 # COLLECTION_ASSET_EXTENSION = f"https://raw.githubusercontent.com/radiantearth/stac-spec/v{STAC_VERSION}/extensions/collection-assets/examples/example-esm.json"
 DATACUBE_EXTENSION = f"https://raw.githubusercontent.com/radiantearth/stac-spec/v{STAC_VERSION}/extensions/datacube/examples/example-item.json"
 EO_EXTENSION = f"https://raw.githubusercontent.com/radiantearth/stac-spec/v{STAC_VERSION}/extensions/eo/examples/example-landsat8.json"
-ITEM_ASSET_EXTENSION = f"https://github.com/radiantearth/stac-spec/blob/v{STAC_VERSION}/extensions/item-assets/examples/example-landsat8.json"
+ITEM_ASSET_EXTENSION = f"https://raw.githubusercontent.com/radiantearth/stac-spec/v1.0.0-beta.2/extensions/item-assets/examples/example-landsat8.json"
```
Comment (geospatial-jeff) [inline tests/test_models.py:42]:
```suggestion
ITEM_ASSET_EXTENSION = f"https://raw.githubusercontent.com/radiantearth/stac-spec/v{STAC_VERSION}/extensions/item-assets/examples/example-landsat8.json"
```

Comment (kylebarron) [reply tests/test_models.py]: Thanks I missed that

**Pull request merged**